lib/spack/spack/build_systems/python.py

```diff
@@ -88,14 +109,14 @@ def import_modules(self):
         return modules
 
     @property
-    def skip_modules(self):
+    def skip_modules(self) -> Iterable[object]:
```

Comment:
Why not:
```suggestion
    def skip_modules(self) -> List[str]:
```
given the docstring?

Reply:
I wanted to make the types as loose as possible. For example, if someone uses a tuple instead of a list, there's no reason for mypy to yell at them, it will work perfectly fine.

Reply:
I would be more specific here. If we know this returns a list of strings we can:
```python
result = self.skip_modules
result.append(...)
result.sort()
filtered = [x for x in result if x.endswith("foo")]
```
My understanding is that, if we use object, we have to:
```python
result = list(self.skip_modules)
result.append(...)
result.sort()
filtered = [x for x in result if str(x).endswith("foo")]
```
I would be glad if `mypy` yells at me, if in the end the result is a more homogeneous interface for users.

(To be clear, this is not a hard request, but just a discussion. Using `str` and `List` is in my opinion also justified by the current implementation of these methods in the builtin repository)

Reply:
We _can_, but we don't. At the moment, the only requirements for `skip_modules` is that it can be iterated over and it can be converted to a string. If someone wants to use a tuple of bytes, I don't see why we should stop them.

Reply:
You might be able to talk me into `Iterable[str]` or `Sequence[str]` but I don't think there's any reason to disallow a tuple.

Reply:
> You might be able to talk me into Iterable[str] or Sequence[str] but I don't think there's any reason to disallow a tuple.

I'm fine with the compromise, but note that using your point from below, there's not a single package that returns a tuple :wink: